daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/FNFunctions.scala

```diff
@@ -858,8 +858,8 @@ case class DFDLXRightShift(recipes: List[CompiledDPath], argType: NodeInfo.Kind)
       case NodeInfo.Int => arg1.getInt >> shift
       case NodeInfo.Short => (arg1.getShort >> shift).toShort
       case NodeInfo.Byte => (arg1.getByte >> shift).toByte
-      case NodeInfo.UnsignedLong =>arg1.getBigInt.shiftRight(shift)
-      case NodeInfo.UnsignedInt => ULong(arg1.getLong.toInt >> shift).toLong
+      case NodeInfo.UnsignedLong =>(ULong(arg1.getBigInt.longValue()) << shift).toBigInt
```

Comment:
This is left shiftt instead of right shift, copy/paste error.

Reply:
A test should definitely have caught this. Make sure there is a test that would fail this.